src/fairness/writer/data_writer_stdout.hpp

```diff
@@ -0,0 +1,73 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#include <sqlite3.h>
+#include <cerrno>
+
+#include "src/fairness/weighted_tree/weighted_walk.hpp"
+#include "src/fairness/reader/data_reader_db.hpp"
+#include "src/fairness/writer/data_writer_base.hpp"
+
+using namespace Flux::accounting;
+
+namespace Flux {
+namespace writer {
+
+/*!  Base data writer class.
+ */
+class data_writer_stdout_t : public data_writer_base_t {
+public:
+    virtual ~data_writer_stdout_t () = default;
+
+    /*! Write fairshare values from a weighted tree.
+     *
+     * \param path          path to a flux-accounting database
+     * \param node          node of the weighted tree of the bank/user hierarchy
+     */
+    int write_acct_info (const std::string &path,
+                         std::shared_ptr<weighted_tree_node_t> node);
+
+    std::string indent = "";
+    bool parsable = false;
+    std::string delimiter = "";
```

Comment:
Rather than making these publically modifiable attributes, maybe make them private and then override the default constructor to take these as arguments and set them within the constructor?

Reply:
Thanks for pointing this out @SteVwonder - I wanted to make sure I understood your suggestion correctly (apologies if I didn't), so I pushed up a couple of `fixup` commits. If they look good, then I can go ahead and squash them. Basically, I moved those attributes to `private` and added a new constructor with parameters that will override the default:

```c++
data_writer_stdout_t (std::string indent = "",
                      bool parsable = false,
                      std::string delimiter = "")
{
    m_indent = indent;
    m_parsable = parsable;
    m_delimiter = delimiter;
}
```

Then, in other places of code where I declare a `data_writer_stdout` object (like in `flux_shares.cpp`), I would initialize the object with these attributes. 

```c++
data_writer_stdout_t data_writer (indent, parsable, delimiter);
```

Hopefully I understood what you were recommending.

Reply:
Yeah. That looks great! Thanks!

Reply:
Woohoo! Thanks. 🙂 I'll set MWP then.